src/index.ts

```diff
@@ -27,6 +32,13 @@ export interface IDockerComposeResult {
   err: string
 }
 
+export type TypedDockerComposeResult<T> = {
+  exitCode: number | null
+  out: string
+  err: string
+  result: T
```

Comment:
@Steveb-p @LarsKumbier I'm thinking about different name than `result`. 

Maybe `data` or `details`?

I think it's pretty common to name the result of the `port` command `result` already which may then lead to `result.result.port` calls as can be seen here: https://github.com/PDMLab/docker-compose/pull/143/files#diff-ec61822d929089017a892b4b047229377c9241f16b43252ab851ae39131cdf40R375

What are your thoughts here?

Reply:
`data` would be ok for me, as it represents the output being parsed by library.